packages/renderer/src/lib/pod/KubernetesTerminalBrowser.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+import type { PodInfoUI } from './PodInfoUI';
+import EmptyScreen from '../ui/EmptyScreen.svelte';
+import NoLogIcon from '../ui/NoLogIcon.svelte';
+import { podsInfos } from '/@/stores/pods';
+import KTerminal from './KubernetesTerminal.svelte';
+import { onMount } from 'svelte';
+
+let key = 0;
+
+export let pod: PodInfoUI;
+
+$: currentContainerStatus =
+  $podsInfos
+    .find(p => p.Name === pod.name)
+    ?.Containers.reduce((acc, c) => {
+      acc.set(c.Names, c.Status);
+      return acc;
+    }, new Map<string, string>()) || new Map<string, string>();
+
+let terminalCache = new Map();
```

Comment:
Limited testing, but this seems to never get cleaned up? Having the history and being able to jump back to the last terminal without losing anything is excellent, but I worry this has a reference to large components and could constantly grow over time. Could you confirm that's not the case, or maybe add some simple clearing like an onDispose() that only keeps e.g. ones that have been touched recently or the last 5?

Reply:
revised a bit this mechanism, included additional checkings for container status update and pod removing, screencast provided below